Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -97,10 +102,31 @@ AdvancedBusEventHandlers advancedBusEventHandlers
 
 
         /// <inheritdoc />
-        public bool IsConnected => connection.IsConnected;
+        public bool IsConnected(PersistentConnectionType type)
+        {
+            return type switch
+            {
+                PersistentConnectionType.Producer => producerConnection.IsConnected,
+                PersistentConnectionType.Consumer => consumerConnection.IsConnected,
+                _ => throw new ArgumentOutOfRangeException(nameof(type), type, null)
+            };
+        }
 
         /// <inheritdoc />
-        public void Connect() => connection.Connect();
+        public void Connect(PersistentConnectionType type)
```

Comment:
Maybe add on overload `Connect()` would be better, because usually you will end up firing both connections, and it's ok to leave also the overload to decide, but the simple Connect() I would add it, what do you think?

Reply:
but then I don't know how to manage the `IsConnected` property/method

Reply:
Let's add both overload for is connected and connect

Reply:
ok I've seen that were already a conversation regarding this point https://github.com/EasyNetQ/EasyNetQ/pull/1273#discussion_r713967257

so i resolve this comment to continue on that conversation